src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/Authentication/RemoteAppAuthenticationService.cs

```diff
@@ -25,70 +25,51 @@ internal partial class RemoteAppAuthenticationService : IRemoteAppAuthentication
         IAuthenticationResultFactory resultFactory,
-        IOptionsSnapshot<RemoteAppAuthenticationClientOptions> authOptions,
+        IOptionsMonitor<RemoteAppAuthenticationClientOptions> authOptions,
         IOptions<RemoteAppClientOptions> remoteAppOptions,
         ILogger<RemoteAppAuthenticationService> logger)
     {
         _resultFactory = resultFactory ?? throw new ArgumentNullException(nameof(resultFactory));
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-        _authOptionsSnapshot = authOptions ?? throw new ArgumentNullException(nameof(authOptions));
+        _authOptionsMonitor = authOptions ?? throw new ArgumentNullException(nameof(authOptions));
         _client = remoteAppOptions?.Value.BackchannelClient ?? throw new ArgumentNullException(nameof(remoteAppOptions));
     }
 
-    /// <summary>
-    /// Initialize the remote authentication service for a given authentication scheme.
-    /// </summary>
-    /// <param name="scheme">The scheme whose configuration should be used to authenticate.</param>
-    public Task InitializeAsync(AuthenticationScheme scheme)
-    {
-        // Finish initializing the http client here since the scheme won't be known
-        // until the owning authentication handler is initialized.
-        _options = _authOptionsSnapshot.Get(scheme.Name);
-
-        return Task.CompletedTask;
-    }
-
     /// <summary>
     /// Authenticate the user of a request by making a request to a remote app using configured
     /// headers and/or cookies from the current request.
     /// </summary>
-    /// <param name="originalRequest">The HTTP request to authenticate.</param>
+    /// <param name="request">The HTTP request to authenticate.</param>
     /// <param name="cancellationToken">A cancellation token.</param>
     /// <returns>
-    /// An authenticaiton result including the claims principal of the authenticated user (if any)
+    /// An authentication result including the claims principal of the authenticated user (if any)
     /// and a status code and headers to include in the response to the request in case the user could not be authenticated.
     /// </returns>
     /// <exception cref="InvalidOperationException">
     /// An invalid operation exception is thrown if
     /// AuthenticateAsync is called without calling Initialize.
     /// </exception>
-    public async Task<RemoteAppAuthenticationResult> AuthenticateAsync(HttpRequest originalRequest, CancellationToken cancellationToken)
+    public async Task<RemoteAppAuthenticationResult> AuthenticateAsync(AuthenticationScheme scheme, HttpRequest request, CancellationToken cancellationToken)
     {
-        if (_options is null)
-        {
-            LogHandlerNotInitialized();
-            throw new InvalidOperationException("Remote authentication handler must be initialized before authenticating");
-        }
+        var options = _authOptionsMonitor.Get(scheme.Name);
```

Comment:
We should probably have some error checking on options. I could imagine options not including the requested scheme.

Reply:
an object will always be returned. however, the fields may not be supplied. however, we're using data annotations on this type so it'll check that for us